### ISSUE TITLE
End of support for Compose V1 

### ISSUE THREAD

**Issue by Madscientiste** [knadh/listmonk#1431]
Using the `install-prod.sh` script on a clean installed server with latest docker fails because it checks for docker-compose command which doesn't not exist for my version (`Docker version 24.0.2, build cb74dfc`).

**Describe the solution you'd like**
replace all `docker-compose` by docker compose.

Should i open a PR for that ? `install-demo.sh` would also need to be updated

**Comment by knadh** [comment]
@mr-karan it's okay to switch away from `docker-compose`, right?

**Comment by mr-karan** [comment]
Yes, `docker-compose`  [is deprecated since June 2023](https://www.docker.com/blog/new-docker-compose-v2-and-v1-deprecation/).

@Madscientiste PR is appreciated.

**Comment by MaximilianKohler** [comment]
@mr-karan does this mean that all instances of `docker-compose` in the docs https://listmonk.app/docs/ need to be changed to `docker compose`? Including `apt install postgresql docker docker-compose`?

**Comment by mr-karan** [comment]
Only the `docker-compose` CLI needs to be changed to `docker compose`. 

> I think apt install postgresql docker docker-compose gets changed to apt install postgresql docker docker-compose-plugin?

`docker compose` is bundled along with `docker` package. Specific packages that need to be installed are listed here: https://docs.docker.com/engine/install/ubuntu/#install-docker-engine

BTW, we don't have specific install instructions for Docker, the same way we don't have for PostgreSQL either. It's upto the users how they install and configure these environments before running Listmonk. 

> And no changes to docker-compose.yml name or content right?

Nope.

**Comment by MaximilianKohler** [comment]
>docker compose is bundled along with docker package. Specific packages that need to be installed are listed here: https://docs.docker.com/engine/install/ubuntu/#install-docker-engine

Are you sure about that? According to that link you have to run `sudo apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin`. So they are explicitly installing `docker-compose-plugin` as well. 

>BTW, we don't have specific install instructions for Docker, the same way we don't have for PostgreSQL either. It's upto the users how they install and configure these environments before running Listmonk.

Yes, I see that now. I patched together [a guide](https://gist.github.com/MaximilianKohler/e5158fcfe6de80a9069926a67afcae11) for newbies ([now linked in the docs](https://listmonk.app/docs/installation/#tutorials)) with a lot of difficulty, and by asking people. I also added a bunch of docker commands to the docs for various basic usage, which I patched together from the github issues. I've been trying to improve the docs so that other people don't have to struggle with everything like I did. I think I will add the docker & postgresql install commands to the docs as well. 

Everyone should be able to use listmonk without being a linux expert. Even @knadh said he's not familiar with docker commands, so it shouldn't be expected that everyone who wants to use listmonk will/should be. There needs to be basic install commands that anyone can use. Experts can always customize the commands themselves. 

It sounds like you're not interested in updating the docs? I'll try to do it and hopefully you can look over the pull request to verify it. 

I'm going to change `sudo apt install postgresql docker docker-compose` to `sudo apt install postgresql docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin`. Hopefully that's correct...

**Comment by mr-karan** [comment]
I've sent PR for replacing `docker-compose` with `docker compose` in the existing docs and installation scripts. 

My point was that since we never had installation docs for docker (arguably, it's the user's preference to run with docker - so the expectation is that docker will be set up on host as well).

I've seen very few projects which actually show installation steps of the dependencies in their own docs. This is what I personally believe but @knadh can take the final call here. Anyway, that should be opened as a new issue as it's unrelated to the `docker-compose` / `docker compose` changes.
